ecl2df/gruptree.py

```diff
@@ -225,10 +223,44 @@ def edge_dataframe2dict(dframe):
 
     children, parents = zip(*edges)
     roots = set(parents).difference(children)
-    trees = []
-    for root in list(roots):
-        trees.append({root: subtrees[root] for root in roots})
-    return trees
+    return [{root: subtrees[root]} for root in sorted(roots)]
+
+
+def _sub_tree_from_dict(nested_dict, name):
```

Comment:
This function creates a new tree instance for each child and pastes it onto the parent. Although this works, another possible approach is to follow along as the treelib documentation shows:

```
>>> tree = Tree()
>>> tree.create_node("Harry", "harry")  # root node
>>> tree.create_node("Jane", "jane", parent="harry")
>>> tree.create_node("Bill", "bill", parent="harry")
>>> tree.create_node("Diane", "diane", parent="jane")
```

If you go for a top-down recursive function instead, only a single tree is created:

```
def _add_to_tree_from_dict(nested_dict, name, tree, parent=None):
    assert isinstance(nested_dict, dict)

    tree.create_node(name, name, parent=parent)
    for key, value in sorted(nested_dict.items()):
        _add_to_tree_from_dict(nested_dict=value, name=key, tree=tree, parent=name)

def tree_from_dict(nested_dict):
    ....
    root_name = list(nested_dict.keys())[0]
    tree = treelib.Tree()
    _add_to_tree_from_dict(nested_dict[root_name], root_name, tree)
    return tree
```

There are different opinions about top-down recursive functions, and having to drag along an instance might not be the most suitable option. But I think I would favour only creating the tree once over the downside of having to pass it down.  I also think the code reads a bit better with this approach.

Reply:
Thanks, agree this looks better.